src/Passes/docs/src/UserGuide/BuildingLibrary.md

```diff
@@ -0,0 +1,149 @@
+# Build guide
+
+This document will guide you through all the requirements and steps that you need to follow to build the QIR adaptor tool (QAT). The first section of this guide helps you install the tools needed to build QAT. After you have finished with the second section of the guide, you should have the `qat` executable compiled and ready to run. As a part of the third part of this post, we will show you how to build and serve all of the documentation including the user guides, developer guides and API documentation. We will go through the steps of compiling the whole library as well as how to run all the tests suite that has been provided as the fourth part of this document. This part is only relevant if you intend to extend or modify the core codebase. There is no need for it when simply building an extension to QAT without having the need to change anything else in the core library. There is a lot more information about how to create a custom extension within the developers’ section.
+
+## Prerequisites
+
+It is necessary for you to have the following tools before you get started on making a new build from this project:
+
+- C++ compiler
+- CMake
+
+If you are compiling and using the command-line tool, these are necessary tools regardless of whether you plan to develop the library itself. The build configuration is handled by CMake, which informs you if a compiler needs to be installed.
+
+### On OS X
+
+Those using OS X can install CMake through the use of a command-line tool called `brew`, as follows:
+
+```sh
+brew install cmake
+```
+
+It should not even be necessary to install anything else on top of OS X, since the compiler comes preinstalled. There is no need to read the next subsection ([Developer prerequisites](#developer-prerequisites)) if you are not intending to modify the core library and can safely jump to the section [Library Dependencies](#library-dependencies)
+
+### On Ubuntu 20.04
+
+Additionally to CMake, you will also require Clang in order to build Ubuntu 20.04. You can accomplish this by following the steps below:
+
+```sh
+apt install clang-11 cmake
+```
+
+You can skip straight to [Library Dependencies](#library-dependencies) if you do not intend to work on developing the core library.
+
+## Developer prerequisites
+
+Development of the core library requires additional tools. These are used formatting, linting and managing code quality. For this purpose, the following tools and packages are used:
+
+- Python 3
+- Python packages specified in `requirements.txt`
+- clang-format
+- clang-tidy
```

Comment:
It may be helpful to provide a `dev-environment.yml` for `conda` that depends on all four of these items.

Reply:
I am not familiar with conda - is that the Windows package manager?

Reply:
Some of your build scripts had called out to `conda`, so I was under the impression that was in-scope, sorry. Regardless, `conda` is a multi-language cross-platform package manager used to create virtual environments containing scientific development tooling. In this case, for example, an environment file like the one below could provide the above prerequisites cross-platform.

```yaml
# dev-environment.yml
name: qat-dev
channels:
- conda-forge
dependencies:
- python>=3.6
- click # Anything else from requirements.txt could go here, too,
        # if there's a matching conda package.
- clang
- clang-tools # Provides clang-format and clang-tidy
- pip
- pip:
  - ... # Anything from requirements.txt that doesn't have a
        # matching conda package can go here.
```

(As for Windows-specific package management, you may be thinking of `choco` or `winget`?)

Reply:
Thanks for the explanation and for the configuration. I am happy to add this.

Yes, it conda was used in the build script by suggestion of @idavis as a way to resolve issues getting LLVM and CMake to play nicely together. I did try `choco` for some other dependencies, but in the end, we never managed to get things to work. It is on my todo list as well.

Reply:
Added